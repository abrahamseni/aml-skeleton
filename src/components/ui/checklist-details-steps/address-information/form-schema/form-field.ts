import { AddressModel } from '@reapit/foundations-ts-definitions'

export interface ValuesType {
  primaryAddress: AddressModel
  secondaryAddress: AddressModel
  metadata: {
    primaryAddress: {
      documentImage?: string
      documentType?: string
      month?: string
      year?: string
    }
    secondaryAddress: {
      documentImage?: string
      documentType?: string
      month?: string
      year?: string
    }
  }
}

export interface FormFieldType {
  typeField: {
    name: 'primaryAddressTypeField' | 'secondaryAddressTypeField'
    label: string
  }
  buildingNameField: {
    name: 'primaryAddress.buildingName' | 'secondaryAddress.buildingName'
    label: string
  }
  buildingNumberField: {
    name: 'primaryAddress.buildingNumber' | 'secondaryAddress.buildingNumber'
    label: string
  }
  line1Field: {
    name: 'primaryAddress.line1' | 'secondaryAddress.line1'
    label: string
  }
  line2Field: {
    name: 'primaryAddress.line2' | 'secondaryAddress.line2'
    label: string
  }
  line3Field: {
    name: 'primaryAddress.line3' | 'secondaryAddress.line3'
    label: string
  }
  line4Field: {
    name: 'primaryAddress.line4' | 'secondaryAddress.line4'
    label: string
  }
  postcodeField: {
    name: 'primaryAddress.postcode' | 'secondaryAddress.postcode'
    label: string
  }
  yearField: {
    name: 'metadata.primaryAddress.year' | 'metadata.secondaryAddress.year'
    label: string
  }
  monthField: {
    name: 'metadata.primaryAddress.month' | 'metadata.secondaryAddress.month'
    label: string
  }
  documentTypeField: {
    name: 'metadata.primaryAddress.documentType' | 'metadata.secondaryAddress.documentType'
    label: string
  }
  documentImageField: {
    name: 'metadata.primaryAddress.documentImage' | 'metadata.secondaryAddress.documentImage'
    label: string
  }
}

export const formFields = (formType: 'primaryAddress' | 'secondaryAddress'): FormFieldType => ({
  typeField: {
    name: `${formType}TypeField`,
    label: 'Type',
  },
  buildingNameField: {
    name: `${formType}.buildingName`,
    label: 'Building Name',
  },
  buildingNumberField: {
    name: `${formType}.buildingNumber`,
    label: 'Building Number',
  },
  line1Field: {
    name: `${formType}.line1`,
    label: 'Line 1',
  },
  line2Field: {
    name: `${formType}.line2`,
    label: 'Line 2',
  },
  line3Field: {
    name: `${formType}.line3`,
    label: 'Line 3',
  },
  line4Field: {
    name: `${formType}.line4`,
    label: 'Line 4',
  },
  postcodeField: {
    name: `${formType}.postcode`,
    label: 'Post Code',
  },
  yearField: {
    name: `metadata.${formType}.year`,
    label: 'Number of Years at Address',
  },
  monthField: {
    name: `metadata.${formType}.month`,
    label: 'Number of Months at Address',
  },
  documentTypeField: {
    name: `metadata.${formType}.documentType`,
    label: 'Document Type',
  },
  documentImageField: {
    name: `metadata.${formType}.documentImage`,
    label: 'Upload File',
  },
})
