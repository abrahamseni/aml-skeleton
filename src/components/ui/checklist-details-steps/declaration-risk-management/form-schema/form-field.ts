export interface ValuesType {
  declarationForm?: string
  reason?: string
  riskAssessmentForm?: string
  type?: string
}

export type AvailableFormFieldType = 'declarationForm' | 'type' | 'reason' | 'riskAssessmentForm'

export interface FormFieldType {
  declarationFormField: {
    name: Extract<AvailableFormFieldType, 'declarationForm'>
    label: string
  }
  typeField: {
    name: Extract<AvailableFormFieldType, 'type'>
    label: string
  }
  reasonField: {
    name: Extract<AvailableFormFieldType, 'reason'>
    label: string
  }
  riskAssessmentFormField: {
    name: Extract<AvailableFormFieldType, 'riskAssessmentForm'>
    label: string
  }
}

export const formField = (): FormFieldType => ({
  declarationFormField: {
    name: 'declarationForm',
    label: 'Upload Declaration File',
  },
  typeField: {
    name: 'type',
    label: 'Risk Assessment Type',
  },
  reasonField: {
    name: 'reason',
    label: 'Reason for Type',
  },
  riskAssessmentFormField: {
    name: 'riskAssessmentForm',
    label: 'Upload Risk Assessment File',
  },
})
