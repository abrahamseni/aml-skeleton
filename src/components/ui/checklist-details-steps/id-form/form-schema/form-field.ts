type FormFieldInfo = {
  name: keyof ValuesType
  label: string
}

export interface ValuesType {
  idType: string
  idReference: string
  expiryDate: string
  documentFile: string
}

export interface FormFieldType {
  idType: FormFieldInfo
  idReference: FormFieldInfo
  expiryDate: FormFieldInfo
  documentFile: FormFieldInfo
}

export const formFields: FormFieldType = {
  idType: {
    name: 'idType',
    label: 'ID Type',
  },
  idReference: {
    name: 'idReference',
    label: 'ID Reference',
  },
  expiryDate: {
    name: 'expiryDate',
    label: 'Expiry Date',
  },
  documentFile: {
    name: 'documentFile',
    label: 'Upload Document',
  },
}
