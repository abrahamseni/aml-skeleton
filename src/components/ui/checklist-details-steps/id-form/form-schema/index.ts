export * from './form-field'
export * from './validation-schema'
