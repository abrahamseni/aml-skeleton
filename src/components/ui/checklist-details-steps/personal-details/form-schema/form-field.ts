type FormFieldInfo = {
  name: string
  label: string
}

export interface ValuesType {
  title?: string
  forename?: string
  surname?: string
  dateOfBirth?: string
  email?: string
  home?: string
  mobile?: string
  work?: string
}

interface PersonalDetailsFormInfo extends FormFieldInfo {
  name: keyof ValuesType
}

export interface FormFieldType {
  title: PersonalDetailsFormInfo
  forename: PersonalDetailsFormInfo
  surname: PersonalDetailsFormInfo
  dateOfBirth: PersonalDetailsFormInfo
  email: PersonalDetailsFormInfo
  homePhone: PersonalDetailsFormInfo
  mobilePhone: PersonalDetailsFormInfo
  workPhone: PersonalDetailsFormInfo
}

export const formFields: FormFieldType = {
  title: {
    name: 'title',
    label: 'Title',
  },
  forename: {
    name: 'forename',
    label: 'Forename',
  },
  surname: {
    name: 'surname',
    label: 'Surname',
  },
  dateOfBirth: {
    name: 'dateOfBirth',
    label: 'Date Of Birth',
  },
  email: {
    name: 'email',
    label: 'Email',
  },
  homePhone: {
    name: 'home',
    label: 'Home Phone',
  },
  mobilePhone: {
    name: 'mobile',
    label: 'Mobile Phone',
  },
  workPhone: {
    name: 'work',
    label: 'Work Phone',
  },
}
