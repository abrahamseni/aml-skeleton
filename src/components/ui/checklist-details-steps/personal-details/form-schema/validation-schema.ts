/* eslint-disable no-confusing-arrow */
import * as Yup from 'yup'
import { errorMessages } from '../../../../../constants/error-messages'
// import { ValuesType } from './form-field'

const PHONE_NO_REGEX = /^[0-9\- ]{8,14}$/

// const validationSchema: Yup.SchemaOf<ValuesType>
const validationSchema = Yup.object().shape({
  title: Yup.string().trim().required(errorMessages.FIELD_REQUIRED),
  forename: Yup.string().trim().required(errorMessages.FIELD_REQUIRED),
  surname: Yup.string().trim().required(errorMessages.FIELD_REQUIRED),
  dateOfBirth: Yup.string().nullable().required(errorMessages.FIELD_REQUIRED),
  email: Yup.string().trim().email('Please enter a valid email format!').required(errorMessages.FIELD_REQUIRED),
  mobilePhone: Yup.string()
    .matches(PHONE_NO_REGEX, {
      message: 'Please enter a valid phone number format',
      excludeEmptyString: true,
    })
    .nullable(),
  workPhone: Yup.string()
    .matches(PHONE_NO_REGEX, {
      message: 'Please enter a valid phone number format',
      excludeEmptyString: true,
    })
    .nullable(),
  homePhone: Yup.string()
    .matches(PHONE_NO_REGEX, {
      message: 'Please enter a valid phone number format',
      excludeEmptyString: true,
    })
    .nullable(),
  // .when(['mobilePhone', 'workPhone'], {
  //   is: (mobilePhone, workPhone) => {
  //     // console.log(!mobilePhone, !workPhone, {})
  //     return !mobilePhone && !workPhone
  //   },
  //   then: (schema) => {
  //     //true condition
  //     // console.log('then', { schema })
  //     return schema.required(errorMessages.FIELD_REQUIRED)
  //   },
  //   otherwise: (schema) => {
  //     //false condition
  //     // console.log('otherwise', { schema })
  //     return schema
  //   },
  // }),
})

export default validationSchema
