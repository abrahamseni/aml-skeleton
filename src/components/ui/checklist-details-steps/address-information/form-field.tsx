import React, { FC, ReactElement, useState } from 'react'
import { InputGroup, Label, Select, InputWrap, InputWrapFull, FlexContainer } from '@reapit/elements'
import { formFields, ValuesType } from './form-schema'
import { UseFormReturn } from 'react-hook-form'
import { generateLabelField, generateOptionsType, generateOptionsYearsOrMonths } from 'utils/generator'
import { cx } from '@linaria/core'
import { order0 } from './__styles__'

import { FileInput } from 'components/ui/ui/file-input'
import DocumentPreviewModal from 'components/ui/ui/document-preview-modal'
import ErrorMessage from 'components/ui/ui/error-message'

interface FormFieldProps {
  name: 'primaryAddress' | 'secondaryAddress'
  useFormProps: {
    errors: UseFormReturn<ValuesType>['formState']['errors']
    register: UseFormReturn<ValuesType>['register']
    getValues: UseFormReturn<ValuesType>['getValues']
  }
}

const FormField: FC<FormFieldProps> = ({ name, useFormProps }): ReactElement => {
  const isPrimaryAddress = name === 'primaryAddress'

  const [imagePrimaryAddress, setImagePrimaryAddress] = useState(false)
  const [imageSecondaryAddress, setImageSecondaryAddress] = useState(false)

  // local function - modal handler
  const modalHandler = (option: 'open' | 'close'): void => {
    if (name === 'primaryAddress') {
      setImagePrimaryAddress(option === 'open')
    } else {
      setImageSecondaryAddress(option === 'open')
    }
  }

  const { register, getValues, errors } = useFormProps

  const {
    typeField,
    buildingNameField,
    buildingNumberField,
    line1Field,
    line2Field,
    line3Field,
    line4Field,
    postcodeField,
    documentImageField,
    monthField,
    yearField,
    documentTypeField,
  } = formFields(name)

  return (
    <>
      <InputWrapFull>
        <InputWrap>
          <InputGroup type="hidden" {...register(typeField.name)} />
        </InputWrap>
        <InputWrap>
          <InputGroup
            type="text"
            autoComplete="off"
            label={generateLabelField(buildingNameField.label)}
            data-testid={buildingNameField.name}
            {...register(buildingNameField.name)}
          />
          <ErrorMessage name={buildingNameField.name} errors={errors} />
        </InputWrap>
        <InputWrap>
          <InputGroup
            type="text"
            autoComplete="off"
            label={generateLabelField(buildingNumberField.label)}
            data-testid={buildingNumberField.name}
            {...register(buildingNumberField.name)}
          />
          <ErrorMessage name={buildingNumberField.name} errors={errors} />
        </InputWrap>
        <InputWrap>
          <InputGroup
            type="text"
            autoComplete="off"
            label={generateLabelField(postcodeField.label, isPrimaryAddress)}
            data-testid={postcodeField.name}
            {...register(postcodeField.name)}
          />
          <ErrorMessage name={postcodeField.name} errors={errors} />
        </InputWrap>
      </InputWrapFull>
      <InputWrapFull>
        <InputWrap>
          <InputGroup
            type="text"
            autoComplete="off"
            label={generateLabelField(line1Field.label, isPrimaryAddress)}
            data-testid={line1Field.name}
            {...register(line1Field.name)}
          />
          <ErrorMessage name={line1Field.name} errors={errors} />
        </InputWrap>
        <InputWrap>
          <InputGroup
            type="text"
            autoComplete="off"
            label={generateLabelField(line2Field.label)}
            data-testid={line2Field.name}
            {...register(line2Field.name)}
          />
          <ErrorMessage name={line2Field.name} errors={errors} />
        </InputWrap>
        <InputWrap>
          <InputGroup
            type="text"
            autoComplete="off"
            label={generateLabelField(line3Field.label, isPrimaryAddress)}
            data-testid={line3Field.name}
            {...register(line3Field.name)}
          />
          <ErrorMessage name={line3Field.name} errors={errors} />
        </InputWrap>
        <InputWrap>
          <InputGroup
            type="text"
            autoComplete="off"
            label={generateLabelField(line4Field.label)}
            data-testid={line4Field.name}
            {...register(line4Field.name)}
          />
          <ErrorMessage name={line4Field.name} errors={errors} />
        </InputWrap>
      </InputWrapFull>
      <InputWrapFull data-testid="option.field.wrapper">
        <InputWrap>
          <InputGroup>
            <Label>{generateLabelField(yearField.label, isPrimaryAddress)}</Label>
            <Select {...register(yearField.name)} data-testid={yearField.name}>
              {generateOptionsYearsOrMonths('years').map((v) => {
                return (
                  <option key={v.value} value={v.value}>
                    {v.label}
                  </option>
                )
              })}
            </Select>
            <ErrorMessage name={yearField.name} errors={errors} />
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Label>{generateLabelField(monthField.label, isPrimaryAddress)}</Label>
            <Select {...register(monthField.name)} data-testid={monthField.name}>
              {generateOptionsYearsOrMonths('months').map((v) => {
                return (
                  <option key={v.value} value={v.value}>
                    {v.label}
                  </option>
                )
              })}
            </Select>
            <ErrorMessage name={monthField.name} errors={errors} />
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Label>{generateLabelField(documentTypeField.label, isPrimaryAddress)}</Label>
            <Select {...register(documentTypeField.name)} data-testid={documentTypeField.name}>
              {generateOptionsType('documentType').map((v) => {
                return (
                  <option key={v.value} value={v.value}>
                    {v.label}
                  </option>
                )
              })}
            </Select>
            <ErrorMessage name={documentTypeField.name} errors={errors} />
          </InputGroup>
        </InputWrap>
        <InputWrap className="el-mt4">
          <FlexContainer isFlexColumn className="el-pl3">
            <Label className={cx('el-mb2', order0)}>
              {generateLabelField(documentImageField.label, isPrimaryAddress)}
            </Label>
            <FileInput
              data-testid={documentImageField.name}
              {...register(documentImageField.name)}
              onFileView={() => modalHandler('open')}
              defaultValue={getValues(documentImageField.name)}
            />
            <ErrorMessage name={documentImageField.name} errors={errors} />
          </FlexContainer>
        </InputWrap>
      </InputWrapFull>
      <DocumentPreviewModal
        src={getValues(documentImageField.name)}
        isOpen={name === 'primaryAddress' ? imagePrimaryAddress : imageSecondaryAddress}
        onModalClose={() => modalHandler('close')}
      />
    </>
  )
}

export default FormField
