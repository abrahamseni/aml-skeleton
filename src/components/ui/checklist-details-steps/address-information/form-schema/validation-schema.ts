import * as Yup from 'yup'
import { errorMessages } from '../../../../../constants/error-messages'
import { ValuesType } from './form-field'

export const validationSchema: Yup.SchemaOf<ValuesType> = Yup.object().shape({
  primaryAddress: Yup.object().shape({
    buildingName: Yup.string().notRequired(),
    buildingNumber: Yup.string().notRequired(),
    line1: Yup.string().required(errorMessages.FIELD_REQUIRED),
    line2: Yup.string().notRequired(),
    line3: Yup.string().required(errorMessages.FIELD_REQUIRED),
    line4: Yup.string().notRequired(),
    postcode: Yup.string().required(errorMessages.FIELD_REQUIRED),
  }),
  secondaryAddress: Yup.object().shape({
    buildingName: Yup.string().notRequired(),
    buildingNumber: Yup.string().notRequired(),
    line1: Yup.string().notRequired(),
    line2: Yup.string().notRequired(),
    line3: Yup.string().notRequired(),
    line4: Yup.string().notRequired(),
    postcode: Yup.string().notRequired(),
  }),
  metadata: Yup.object().shape({
    primaryAddress: Yup.object().shape({
      documentImage: Yup.string()
        .required(errorMessages.FIELD_REQUIRED)
        .matches(/(png|jpg|jpeg|pdf)/, errorMessages.WRONG_FILE_TYPE),
      documentType: Yup.string().required(errorMessages.FIELD_REQUIRED),
      month: Yup.string().required(errorMessages.FIELD_REQUIRED),
      year: Yup.string().required(errorMessages.FIELD_REQUIRED),
    }),
    secondaryAddress: Yup.object().shape({
      documentImage: Yup.string().notRequired(),
      documentType: Yup.string().notRequired(),
      month: Yup.string().notRequired(),
      year: Yup.string().notRequired(),
    }),
  }),
})
