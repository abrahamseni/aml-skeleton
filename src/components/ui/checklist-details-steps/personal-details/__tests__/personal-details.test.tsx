/* eslint-disable no-unused-vars */
/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import AxiosMockAdapter from 'axios-mock-adapter'
import { QueryClient, QueryClientProvider } from 'react-query'

import Axios from 'axios/axios'
import { URLS } from 'constants/api'
import { formFields } from '../form-schema/form-field'
import PersonalDetails from '../personal-details'
import { wait } from 'utils/test'
import { error, success } from 'utils/mocks/useSnack'
import { CONTACT_MOCK_DATA_1, CONTACT_MOCK_DATA_2 } from 'platform-api/__mocks__/contact-api.mock'

const axiosMock = new AxiosMockAdapter(Axios, {
  onNoMatch: 'throwException',
})

jest.mock('@reapit/elements', () => jest.requireActual('utils/mocks/reapit-element-mocks'))
jest.unmock('@reapit/connect-session')
jest.mock('../../../../../core/connect-session')

const { title, forename, surname, dateOfBirth, email, workPhone, mobilePhone, homePhone } = formFields

describe('Personal Details Form', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    axiosMock.reset()
  })

  describe('Snapshot', () => {
    it('should match a snapshot', () => {
      expect(renderComponent(defaultPersonalDetailsProps)).toMatchSnapshot()
    })
  })

  describe('DOM', () => {
    it('should render 8 fields', () => {
      const { container } = renderComponent(defaultPersonalDetailsProps)

      const totalTextFields = container.querySelectorAll<HTMLInputElement>('[type="text"]')
      const totalDateFields = container.querySelectorAll<HTMLInputElement>('[type="date"]')
      const totalEmailields = container.querySelectorAll<HTMLInputElement>('[type="email"]')

      expect(totalTextFields.length).toEqual(6)
      expect(totalDateFields.length).toEqual(1)
      expect(totalEmailields.length).toEqual(1)
      expect(totalTextFields.length + totalDateFields.length + totalEmailields.length).toEqual(8)
    })

    it('the fields should able to automatically fill with default value and be able to change', () => {
      const { getByTestId } = renderComponent(defaultPersonalDetailsProps)

      // title
      const testTitleField = getByTestId(`test.${title.name}`) as HTMLInputElement
      expect(testTitleField.value).toMatch(/Mr/i)
      // forename
      const testForenameField = getByTestId(`test.${forename.name}`) as HTMLInputElement
      expect(testForenameField.value).toMatch(/Christopher/i)
      // surname
      const testSurnameField = getByTestId(`test.${surname.name}`) as HTMLInputElement
      expect(testSurnameField.value).toMatch(/Williamson/i)
      // date of birth
      const testDateOfBirthField = getByTestId(`test.${dateOfBirth.name}`) as HTMLInputElement
      expect(testDateOfBirthField.value).toMatch(/1988-05-19/i)
      // email
      const testEmailField = getByTestId(`test.${email.name}`) as HTMLInputElement
      expect(testEmailField.value).toMatch(/cwilliamson47@rpsfiction.net/i)
      // home phone
      const testHomePhoneField = getByTestId(`test.${homePhone.name}`) as HTMLInputElement
      expect(testHomePhoneField.value).toMatch(/02471896/i)
      // mobile phone
      const testMobilePhoneField = getByTestId(`test.${mobilePhone.name}`) as HTMLInputElement
      expect(testMobilePhoneField.value).toMatch(/07110660399/i)
      // work phone
      const testWorkPhoneField = getByTestId(`test.${workPhone.name}`) as HTMLInputElement
      expect(testWorkPhoneField.value).toMatch(/028233444/i)

      // title changed
      fireEvent.change(testTitleField, { target: { value: 'Ms' } })
      expect(testTitleField.value).toMatch(/Ms/i)
      // forename changed
      fireEvent.change(testTitleField, { target: { value: 'Will' } })
      expect(testTitleField.value).toMatch(/Will/i)
      // surname changed
      fireEvent.change(testSurnameField, { target: { value: 'Helliam' } })
      expect(testSurnameField.value).toMatch(/Helliam/i)
      // date of birth changed
      fireEvent.change(testDateOfBirthField, { target: { value: '1989-02-19' } })
      expect(testDateOfBirthField.value).toMatch(/1989-02-19/i)
      // email changed
      fireEvent.change(testTitleField, { target: { value: 'cwilliamson47@reapit.au' } })
      expect(testTitleField.value).toMatch(/cwilliamson47@reapit.au/i)
      // home phone changed
      fireEvent.change(testHomePhoneField, { target: { value: '02471800' } })
      expect(testHomePhoneField.value).toMatch(/02471800/i)
      // mobile phone changed
      fireEvent.change(testMobilePhoneField, { target: { value: '07110880396' } })
      expect(testMobilePhoneField.value).toMatch(/07110880396/i)
      // work phone changed
      fireEvent.change(testWorkPhoneField, { target: { value: '028246444' } })
      expect(testWorkPhoneField.value).toMatch(/028246444/i)
    })

    it('should display error message if required field is empty', async () => {
      const { getByTestId } = renderComponent(defaultPersonalDetailsProps)

      // title
      const testTitleField = getByTestId(`test.${title.name}`) as HTMLTextAreaElement
      expect(testTitleField.value).toMatch(/Mr/i)

      fireEvent.change(testTitleField, { target: { value: '' } })
      await wait(0)
      fireEvent.blur(testTitleField)
      await wait(0)

      const testTitleFieldErrorMessage = getByTestId(`error.${title.name}`)
      expect(testTitleFieldErrorMessage).not.toBeUndefined
      expect(testTitleFieldErrorMessage.textContent).toMatch(/Required/i)
      // forename
      const testForenameField = getByTestId(`test.${forename.name}`) as HTMLTextAreaElement
      expect(testForenameField.value).toMatch(/Christopher/i)

      fireEvent.change(testForenameField, { target: { value: '' } })
      await wait(0)
      fireEvent.blur(testForenameField)
      await wait(0)

      const testForenameFieldErrorMessage = getByTestId(`error.${forename.name}`)
      expect(testForenameFieldErrorMessage).not.toBeUndefined
      expect(testForenameFieldErrorMessage.textContent).toMatch(/Required/i)
      // surname
      const testSurnameField = getByTestId(`test.${surname.name}`) as HTMLTextAreaElement
      expect(testSurnameField.value).toMatch(/Williamson/i)

      fireEvent.change(testSurnameField, { target: { value: '' } })
      await wait(0)
      fireEvent.blur(testSurnameField)
      await wait(0)

      const testSurnameFieldErrorMessage = getByTestId(`error.${surname.name}`)
      expect(testSurnameFieldErrorMessage).not.toBeUndefined
      expect(testSurnameFieldErrorMessage.textContent).toMatch(/Required/i)
      // date of birth
      const testDateOfBirth = getByTestId(`test.${dateOfBirth.name}`) as HTMLTextAreaElement
      expect(testDateOfBirth.value).toMatch(/1988-05-19/i)

      fireEvent.change(testDateOfBirth, { target: { value: '' } })
      await wait(0)
      fireEvent.blur(testDateOfBirth)
      await wait(0)

      const testDateOfBirthErrorMessage = getByTestId(`error.${dateOfBirth.name}`)
      expect(testDateOfBirthErrorMessage).not.toBeUndefined
      expect(testDateOfBirthErrorMessage.textContent).toMatch(/Required/i)
      // email
      const testEmailField = getByTestId(`test.${email.name}`) as HTMLTextAreaElement
      expect(testEmailField.value).toMatch(/cwilliamson47@rpsfiction.net/i)

      fireEvent.change(testEmailField, { target: { value: '' } })
      await wait(0)
      fireEvent.blur(testEmailField)
      await wait(0)

      const testEmailFieldErrorMessage = getByTestId(`error.${email.name}`)
      expect(testEmailFieldErrorMessage).not.toBeUndefined
      expect(testEmailFieldErrorMessage.textContent).toMatch(/Required/i)
    })
  })
})

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },
})

const defaultPersonalDetailsProps = {
  userData: CONTACT_MOCK_DATA_1,
}

const renderComponent = (props, type: 'v1' | 'v2' = 'v1') => {
  const SELECTED_MOCK_CONTACT = type === 'v1' ? CONTACT_MOCK_DATA_1 : CONTACT_MOCK_DATA_2

  axiosMock.onGet(`${URLS.CONTACTS}/${SELECTED_MOCK_CONTACT.id}`).reply(200, SELECTED_MOCK_CONTACT)

  return render(
    <QueryClientProvider client={queryClient}>
      <PersonalDetails {...props} userData={SELECTED_MOCK_CONTACT} />
    </QueryClientProvider>,
  )
}
