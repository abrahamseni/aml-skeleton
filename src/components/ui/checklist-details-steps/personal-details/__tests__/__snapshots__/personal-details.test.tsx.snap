// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal Details Form Snapshot should match a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        data-testid="personal.details.form"
      >
        <div
          class="el-flex el-flex-column el-flex-wrap"
        >
          <div
            class="el-flex1"
          >
            <input
              data-testid="test.title"
              id="title"
              name="title"
              type="text"
            />
            <label
              for="title"
            >
              Title *
            </label>
          </div>
          <div
            class="el-mt6 el-flex1"
          >
            <input
              data-testid="test.forename"
              id="forename"
              name="forename"
              type="text"
            />
            <label
              for="forename"
            >
              Forename *
            </label>
          </div>
          <div
            class="el-mt6 el-flex1"
          >
            <input
              data-testid="test.surname"
              id="surname"
              name="surname"
              type="text"
            />
            <label
              for="surname"
            >
               
              Surname *
            </label>
          </div>
        </div>
        <div
          class="el-mt6 el-flex el-flex-column el-flex-wrap el-w6"
        >
          <div
            class="el-flex1"
          >
            <input
              data-testid="test.dateOfBirth"
              id="dateOfBirth"
              name="dateOfBirth"
              type="date"
            />
            <label
              for="dateOfBirth"
            >
              Date Of Birth *
            </label>
          </div>
          <div
            class="el-mt6 el-flex1"
          >
            <input
              data-testid="test.email"
              id="email"
              name="email"
              type="email"
            />
            <label
              for="email"
            >
              Email *
            </label>
          </div>
        </div>
        <div
          class="el-mt8"
        >
          <p
            class="mock-combined-css-class"
          >
            *At least one telephone number is required
          </p>
          <div
            class=" el-flex el-flex-column el-flex-wrap"
          >
            <div
              class="el-flex1"
            >
              <input
                data-testid="test.homePhone"
                id="homePhone"
                name="homePhone"
                type="text"
              />
              <label
                for="homePhone"
              >
                Home Phone
              </label>
            </div>
            <div
              class="el-mt6 el-flex1"
            >
              <input
                data-testid="test.mobilePhone"
                id="mobilePhone"
                name="mobilePhone"
                type="text"
              />
              <label
                for="mobilePhone"
              >
                Mobile Phone
              </label>
            </div>
            <div
              class="el-mt6 el-flex1"
            >
              <input
                data-testid="test.workPhone"
                id="workPhone"
                name="workPhone"
                type="text"
              />
              <label
                for="workPhone"
              >
                Work Phone
              </label>
            </div>
          </div>
        </div>
        <footer
          class="el-mt8"
        >
          <div
            class="mock-combined-css-class"
          >
            <p
              class="mock-combined-css-class"
            >
              RPS Ref: 
              MLK16000071
            </p>
            <button
              class="mock-combined-css-class"
              data-testid="save-form"
              type="submit"
            >
              Save
            </button>
          </div>
          <div
            class="el-mt6"
          >
            <p
              class="mock-combined-css-class"
            >
              * Indicates fields that are required in order to 'Complete' this section.
            </p>
          </div>
        </footer>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      data-testid="personal.details.form"
    >
      <div
        class="el-flex el-flex-column el-flex-wrap"
      >
        <div
          class="el-flex1"
        >
          <input
            data-testid="test.title"
            id="title"
            name="title"
            type="text"
          />
          <label
            for="title"
          >
            Title *
          </label>
        </div>
        <div
          class="el-mt6 el-flex1"
        >
          <input
            data-testid="test.forename"
            id="forename"
            name="forename"
            type="text"
          />
          <label
            for="forename"
          >
            Forename *
          </label>
        </div>
        <div
          class="el-mt6 el-flex1"
        >
          <input
            data-testid="test.surname"
            id="surname"
            name="surname"
            type="text"
          />
          <label
            for="surname"
          >
             
            Surname *
          </label>
        </div>
      </div>
      <div
        class="el-mt6 el-flex el-flex-column el-flex-wrap el-w6"
      >
        <div
          class="el-flex1"
        >
          <input
            data-testid="test.dateOfBirth"
            id="dateOfBirth"
            name="dateOfBirth"
            type="date"
          />
          <label
            for="dateOfBirth"
          >
            Date Of Birth *
          </label>
        </div>
        <div
          class="el-mt6 el-flex1"
        >
          <input
            data-testid="test.email"
            id="email"
            name="email"
            type="email"
          />
          <label
            for="email"
          >
            Email *
          </label>
        </div>
      </div>
      <div
        class="el-mt8"
      >
        <p
          class="mock-combined-css-class"
        >
          *At least one telephone number is required
        </p>
        <div
          class=" el-flex el-flex-column el-flex-wrap"
        >
          <div
            class="el-flex1"
          >
            <input
              data-testid="test.homePhone"
              id="homePhone"
              name="homePhone"
              type="text"
            />
            <label
              for="homePhone"
            >
              Home Phone
            </label>
          </div>
          <div
            class="el-mt6 el-flex1"
          >
            <input
              data-testid="test.mobilePhone"
              id="mobilePhone"
              name="mobilePhone"
              type="text"
            />
            <label
              for="mobilePhone"
            >
              Mobile Phone
            </label>
          </div>
          <div
            class="el-mt6 el-flex1"
          >
            <input
              data-testid="test.workPhone"
              id="workPhone"
              name="workPhone"
              type="text"
            />
            <label
              for="workPhone"
            >
              Work Phone
            </label>
          </div>
        </div>
      </div>
      <footer
        class="el-mt8"
      >
        <div
          class="mock-combined-css-class"
        >
          <p
            class="mock-combined-css-class"
          >
            RPS Ref: 
            MLK16000071
          </p>
          <button
            class="mock-combined-css-class"
            data-testid="save-form"
            type="submit"
          >
            Save
          </button>
        </div>
        <div
          class="el-mt6"
        >
          <p
            class="mock-combined-css-class"
          >
            * Indicates fields that are required in order to 'Complete' this section.
          </p>
        </div>
      </footer>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
